dali/dfu/dfuwu.cpp

```diff
@@ -2962,6 +3026,57 @@ class CDFUWorkUnitFactory : implements IDFUWorkUnitFactory, implements ISDSSubsc
         // created time stamp? TBD
         return ret;
     }
+
+    void createPublisherWorkUnit(StringBuffer &wuid, const char *jobname, const char *queue) override
+    {
+        newWUID(wuid, 'P');
+        StringBuffer wuRoot;
+        getXPath(wuRoot, wuid.str());
+        Owned<IRemoteConnection> conn = querySDS().connect(wuRoot.str(), myProcessSession(), RTM_LOCK_WRITE|RTM_CREATE_UNIQUE, SDS_LOCK_TIMEOUT);
+        if (!conn)
+            return;
+        OwnedPTree root = conn->getRoot();
+        root->setProp("@command", "publish");
+        if (!isEmptyString(jobname))
+            root->setProp("@jobName", jobname);
+        if (!isEmptyString(queue))
+            root->setProp("@queue", queue);
+        IPropertyTree *progress = root->addPropTree("Progress");
+        progress->setPropInt("Edition", 1);
+        progress->setPropInt("@percentdone", 0);
+        progress->setPropInt("@taskcount", 0);
+        progress->setPropInt("@taskscomplete", 0);
+        progress->setProp("@state", "started");
+        conn->commit();
+    }
+
+    unsigned incrementPublisherTaskCount(const char *parent)
+    {
+        StringBuffer wuProgressRoot;
+        getXPath(wuProgressRoot, parent);
+        wuProgressRoot.append("/Progress");
+        Owned<IRemoteConnection> conn = querySDS().connect(wuProgressRoot.str(), myProcessSession(), RTM_LOCK_WRITE, SDS_LOCK_TIMEOUT);
```

Comment:
Slightly concerned about the overhead of this operation.  Better to hold on to the connection (via a CDfuWorkuUnit?) and increment, or set once all tasks are created.  @jakesmith would you be concerned?